protocol/messenger.go

```diff
@@ -1894,6 +1894,63 @@ func (m *Messenger) LeaveGroupChat(ctx context.Context, chatID string, remove bo
 	return m.leaveGroupChat(ctx, &response, chatID, remove, true)
 }
 
+// Decline all pending group invites from a user
+func (m *Messenger) DeclineAllPendingGroupInvitesFromUser(response *MessengerResponse, userPublicKey string) (*MessengerResponse, error) {
+
+	// Decline group invites from active chats
+	chats, err := m.persistence.Chats()
+	var ctx context.Context
+	if err != nil {
+		return nil, err
+	}
+
+	for _, chat := range chats {
+		if chat.ChatType == ChatTypePrivateGroupChat &&
+			chat.ReceivedInvitationAdmin == userPublicKey &&
+			chat.Joined == 0 && chat.Active {
+			response, err = m.leaveGroupChat(ctx, response, chat.ID, true, true)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// Decline group invites from activity center notifications
+	var ids []types.HexBytes
+	var notifications []*ActivityCenterNotification
+	cursor := ""
+
+	for {
+		cursor, notifications, err = m.persistence.ActivityCenterNotifications(cursor, 10)
```

Comment:
For now is probably ok, but you could move this checks to the database, by querying exactly what you need.